miprometheus/problems/video_text_to_class/cog/cog_utils/generate_dataset.py

```diff
@@ -0,0 +1,281 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+"""Script for generating a COG dataset"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import errno
+import functools
+import gzip
+import itertools
+import json
+import multiprocessing
+import os
+import random
+import shutil
+import traceback
+
+import numpy as np
+import tensorflow as tf
```

Comment:
Ok, so what is really going in here? Seems that you are introducing dependencies on TF - an we really do not want new dependencies... 

After a brief analysis it seems that those are only for processing input arguments. In our workers we are extensively using argparse, please analyze the following abstract worker code:
https://github.com/IBM/mi-prometheus/blob/b4f2442f6a108dd2d546e9df933e0d47ec9e51dd/miprometheus/workers/worker.py#L33
https://github.com/IBM/mi-prometheus/blob/b4f2442f6a108dd2d546e9df933e0d47ec9e51dd/miprometheus/workers/worker.py#L109
https://github.com/IBM/mi-prometheus/blob/b4f2442f6a108dd2d546e9df933e0d47ec9e51dd/miprometheus/workers/worker.py#L182

Is this sufficient or do you need something more?